compiler/rustc_middle/src/mir/interpret/allocation.rs

```diff
@@ -599,6 +605,9 @@ impl<Tag: Copy, Extra> Allocation<Tag, Extra> {
     /// Applies a relocation copy.
     /// The affected range, as defined in the parameters to `prepare_relocation_copy` is expected
     /// to be clear of relocations.
+    ///
+    /// This is dangerous to use as it can violate internal `Allocation` invariants!
+    /// It only exists to support an efficient implementation of `mem_copy_repeatedly`.
```

Comment:
I tried finding a nice safe API we could add here to support `mem_copy_repeatedly` and make all this "compressed range" stuff private, but couldn't come up with anything good.

Reply:
The warnings are there now, so we can look at it again in the future